tree/30_generic_methods/kernel_module_disable_at_boot.cf

```diff
@@ -0,0 +1,54 @@
+# @name Kernel module disabled at boot
+# @description Ensure that a given kernel module is disabled and loaded on the system
+# @documentation Ensure that a given kernel module is disabled and loaded on the system.
+#  If the module is not disabled, Rudder will create its install conf in /etc/modprobe.d/manage_by_rudder.conf.
+#  Which means that it will redirect the module loading to /bin/false, preventing its loading and the load of
+#  any module which depends on this one.
+#
+#
+# @parameter module_name Complete name of the kernel module, as seen by lsmod or listed in /proc/modules
+#
+# @class_prefix kernel_module_disabled_at_boot
+# @class_parameter module_name
+
+
+bundle agent kernel_module_disabled_at_boot(module_name)
+{
+  vars:
+      "modprobe_dir"            string => "/etc/modprobe.d";
+
+      "old_class_prefix"        string => canonify("variable_module_disabled_at_boot_${module_name}");
+      "args"                     slist => { "${module_name}", "${configuration}" };
+      "report_param"            string => join("_", args);
+      "full_class_prefix"       string => canonify("kernel_module_disabled_at_boot_${report_param}");
+      "old_class_prefix"        string => string_head("${full_class_prefix}", "1000");
+      "class_prefix"            string => string_head("${full_class_prefix}", "1000");
+
+      "configuration"           string => "install ${module_name} /bin/true";
+
+      "inner_args"               slist => { "${modprobe_dir}/manage_by_rudder.conf", "### ${module_name} start section", "### ${module_name} end section", "${configuration}" };
```

Comment:
you should probably use variables for all the inner_args, so that we can reuse the variable within the file_block_present_in_section, and ease maintenance